### PR TITLE
Chowtapemodel: unstable-2020-12-12 -> 2.10.0

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/CHOWTapeModel/default.nix [hunk 1]
```diff
@@ -1,62 +1,106 @@
...
+    # util-linuxMinimal
+    # webkitgtk
+    gcc-unwrapped
   ];
 
-  buildPhase = ''
-    cd Plugin/
-    ./build_linux.sh
-  '';
+  cmakeFlags = [ "-DCMAKE_AR=${gcc-unwrapped}/bin/gcc-ar"
```
Comment (symphorien) [inline pkgs/applications/audio/CHOWTapeModel/default.nix:86]:
```suggestion
  cmakeFlags = [ "-DCMAKE_AR=${stdenv.cc.targetPrefix}ar"
```
same for the two other below

**Review by symphorien — CHANGES_REQUESTED** [review]
segfaults at startup for me (x86_64 linux nixos 21.05). Also did you check that open-file dialogs work because this probably needs wrapGappsHook.

File: pkgs/applications/audio/CHOWTapeModel/default.nix [hunk 2]
```diff
@@ -1,62 +1,106 @@
...
+                 "-DCMAKE_RANLIB=${gcc-unwrapped}/bin/gcc-ranlib"
+                 "-DCMAKE_NM=${gcc-unwrapped}/bin/gcc-nm" ];
+
+  postPatch = "cd Plugin";
+
+  # buildPhase = ''
+  #   cmake -Bbuild
+  #   cmake --build build/ --config Release
+  # '';
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/CHOWTapeModel/default.nix:95]: ?

File: pkgs/applications/audio/CHOWTapeModel/default.nix [hunk 3]
```diff
@@ -1,62 +1,106 @@
...
+    libselinux
+    libsepol
+    libsysprof-capture
+    libthai
+    libxkbcommon
+    pcre
     python3
+    sqlite
+    # util-linuxMinimal
+    # webkitgtk
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/CHOWTapeModel/default.nix:82]:
```suggestion
```

File: pkgs/applications/audio/CHOWTapeModel/default.nix (hunk 1 above)
Comment (magnetophon) [reply pkgs/applications/audio/CHOWTapeModel/default.nix]: When I do this, I get lots of linker errors.
Oddly enough, the build output in that case has all 3 flags twice; once with a sensible path to the store, and once without a path, so like this:
```
DCMAKE_AR=ar
```

When I remove these lines all together, I still get the flags with full path, but also the linker errors.
When I put back the original lines (as in the current state of this PR), it all builds just fine.

**Comment by magnetophon** [comment]
> segfaults at startup for me (x86_64 linux nixos 21.05). Also did you check that open-file dialogs work because this probably needs wrapGappsHook.

It did for me as well.
Now it all works here.
For you too?

File: pkgs/applications/audio/CHOWTapeModel/default.nix (hunk 2 above)
Comment (magnetophon) [reply pkgs/applications/audio/CHOWTapeModel/default.nix]: Sorry for the mess; removed now.

**Comment by magnetophon** [comment]
Also, this PR has been updated from version 2.8.0 to 2.10.0.

File: pkgs/applications/audio/CHOWTapeModel/default.nix (hunk 1 above)
Comment (magnetophon) [reply pkgs/applications/audio/CHOWTapeModel/default.nix]: More detailed info [here](https://github.com/NixOS/nixpkgs/pull/136761#discussion_r764816618).

**Comment by applePrincess** [comment]
```
[alice@applePrincess:~/work/nixpkgs]$ nixpkgs-review pr 136762
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/136762/head:refs/nixpkgs-review/1
remote: Enumerating objects: 9, done.
remote: Counting objects: 100% (7/7), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 9 (delta 0), reused 5 (delta 0), pack-reused 2
Unpacking objects: 100% (9/9), 235.34 KiB | 1.49 MiB/s, done.
From https://github.com/NixOS/nixpkgs
   f016dd04c7f..fedde5d24f4  master                -> refs/nixpkgs-review/0
 + fd4322555ae...f6e7906f2fe refs/pull/136762/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/alice/.cache/nixpkgs-review/pr-136762/nixpkgs fedde5d24f4e749a62d3bebcb86e0bb71ce92c64
Preparing worktree (detached HEAD fedde5d24f4)
Updating files: 100% (29391/29391), done.
HEAD is now at fedde5d24f4 woeusb-ng: init at 0.2.10
$ nix-env --option system x86_64-linux -f /home/alice/.cache/nixpkgs-review/pr-136762/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit --no-ff f6e7906f2fe7991cd88661b87176f71b4b30e897
Automatic merge went well; stopped before committing as requested
$ nix-env --option system x86_64-linux -f /home/alice/.cache/nixpkgs-review/pr-136762/nixpkgs -qaP --xml --out-path --show-trace --meta
error: anonymous function at /home/alice/.cache/nixpkgs-review/pr-136762/nixpkgs/pkgs/applications/audio/CHOWTapeModel/default.nix:1:1 called without required argument 'epoxy'

       at /home/alice/.cache/nixpkgs-review/pr-136762/nixpkgs/lib/customisation.nix:69:16:

           68|     let
           69|       result = f origArgs;
             |                ^
           70|

       … while evaluating 'makeOverridable'

       at /home/alice/.cache/nixpkgs-review/pr-136762/nixpkgs/lib/customisation.nix:67:24:

           66|   */
           67|   makeOverridable = f: origArgs:
             |                        ^
           68|     let

       … from call site

       at /home/alice/.cache/nixpkgs-review/pr-136762/nixpkgs/lib/customisation.nix:121:8:

          120|       auto = builtins.intersectAttrs (lib.functionArgs f) autoArgs;
          121|     in makeOverridable f (auto // args);
             |        ^
          122|

       … while evaluating 'callPackageWith'

       at /home/alice/.cache/nixpkgs-review/pr-136762/nixpkgs/lib/customisation.nix:117:35:

          116|   */
          117|   callPackageWith = autoArgs: fn: args:
             |                                   ^
          118|     let

       … from call site

       at /home/alice/.cache/nixpkgs-review/pr-136762/nixpkgs/pkgs/top-level/all-packages.nix:24611:19:

        24610|
        24611|   CHOWTapeModel = callPackage ../applications/audio/CHOWTapeModel { };
             |                   ^
        24612|
https://github.com/NixOS/nixpkgs/pull/136762 failed to build
$ git worktree prune
```

I got this. 
Also just to meniton Takes over #143258 to not to forget to close when this pr is merged

**Comment by magnetophon** [comment]
Hmm, I get the same when I test via nixpkgs-review.
When I check out the branch manually, it builds and runs.

**Comment by applePrincess** [comment]
Ah yeah. interesting...

```
[alice@applePrincess:~/work/nixpkgs]$ git checkout f6e7906f2fe7991cd88661b87176f71b4b30e897
Updating files: 100% (2444/2444), done.
Note: switching to 'f6e7906f2fe7991cd88661b87176f71b4b30e897'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at f6e7906f2fe Chowtapemodel: unstable-2020-12-12 -> 2.10.0

[alice@applePrincess:~/work/nixpkgs]$ nix build .#CHOWTapeModel

[alice@applePrincess:~/work/nixpkgs]$ ./
doc/         .git/        .github/     lib/         maintainers/ nixos/       outputs/     pkgs/        result/      

[alice@applePrincess:~/work/nixpkgs]$ readlink ./result
/nix/store/nwh68dwa7scn22si02s8c6yqs4if5lhd-CHOWTapeModel-2.10.0

[alice@applePrincess:~/work/nixpkgs]$ ./result/bin/CHOWTapeModel 

```

It builds manually.

**Comment by symphorien** [comment]
It's because epoxy is an alias to libepoxy, and aliases are forbidden by CI.

**Comment by magnetophon** [comment]
@symphorien Thanks, fixed now.

**Review by symphorien — APPROVED** [review]

**Pull request merged**